spec/models/in_person_enrollment_spec.rb

```diff
@@ -417,6 +417,41 @@
         expect(enrollment.days_to_due_date).to eq(validity_in_days - days_ago_established_at)
       end
     end
+
+    context 'check edges to confirm date calculation is correct' do
+      it 'returns the correct due date and days to due date with 1 day left' do
+        freeze_time do
+          enrollment = create(
+            :in_person_enrollment,
+            enrollment_established_at: Time.zone.now - 9.days,
```

Comment:
is there a config this is based on that could be used? Ditto for the other places where the tests use 9.5, 10, etc.

Reply:
I could set the values in let blocks like [above](https://github.com/18F/identity-idp/pull/9477/files/fa67fbb33c8d34891e694e25f204ad72300e094e#diff-b43477c280953091d41fc68e83e393c6eb727f0af7c6d40b240ab34dc35e836cR388).

Reply:
@mitchellhenke see [b0fcb1847acc2ffb8b212e7cc55b54e339b71de5](https://github.com/18F/identity-idp/pull/9477/commits/b0fcb1847acc2ffb8b212e7cc55b54e339b71de5)

Reply:
@JackRyan1989 I think @mitchellhenke's point is that the timespans depend on the `in_person_enrollment_validity_in_days` configuration, so we should control that either by basing the timespans on the configuration, or stubbing the configuration to a fixed value for the duration of the test.

The risk is that if we change the default configuration in the future or if a developer has their own overrides for the configuration value, the tests may suddenly start breaking.

Reply:
Ah whoops, I got confused. I thought you were specifically talking about the calculation for the `enrollment_established_at` value, and that can be any date and is not necessarily dependent on the configuration. 

I've updated the expectations to match the previous pattern, e.g. ` expect(enrollment.days_to_due_date).to eq(validity_in_days - days_ago_established_at)`

So for example: 
```
expect(enrollment.days_to_due_date).to eq((validity_in_days - nine).to_i)
  expect(enrollment.due_date).to(
            eq((validity_in_days - nine).days.from_now),
            )
```

I think I understand now, but please let me know if I'm still missing something.

Reply:
Sorry, yeah, I didn't explain it well. The goal would be to avoid a change in `validity_in_days` changing and breaking the tests. fa67fbb33c8d34891e694e25f204ad72300e094e was I think the preferred approach.

My suggestion is something along these lines:

```ruby
    it 'due_date returns the enrollment expiration date based on when it was established' do
      freeze_time do
        enrollment = create(
          :in_person_enrollment,
          enrollment_established_at: 3.days.ago,
        )
        expect(enrollment.due_date).to eq((validity_in_days - 3).days.from_now)
      end
    end

    it 'days_to_due_date returns the number of days left until the due date' do
      freeze_time do
        enrollment = create(
          :in_person_enrollment,
          enrollment_established_at: 3.days.ago,
        )
        expect(enrollment.days_to_due_date).to eq(validity_in_days - 3)
      end
    end

    context 'check edges to confirm date calculation is correct' do
      it 'returns the correct due date and days to due date with 1 day left' do
        freeze_time do
          enrollment = create(
            :in_person_enrollment,
            enrollment_established_at: (validity_in_days - 1).days.ago,
          )
          expect(enrollment.days_to_due_date).to eq(1)
          expect(enrollment.due_date).to eq(Time.zone.now + 1.day)
        end
      end

      it 'returns the correct due date and days to due date with 0.5 days left' do
        freeze_time do
          enrollment = create(
            :in_person_enrollment,
            enrollment_established_at: (validity_in_days - 0.5).days.ago,
          )
          expect(enrollment.days_to_due_date).to eq(0)
          expect(enrollment.due_date).to eq(Time.zone.now + 0.5.days)
        end
      end

      it 'returns the correct due date and days to due date with 0 days left' do
        freeze_time do
          enrollment = create(
            :in_person_enrollment,
            enrollment_established_at: validity_in_days.days.ago,
          )
          expect(enrollment.days_to_due_date).to eq(0)
          expect(enrollment.due_date).to eq(Time.zone.now)
        end
      end
    end

```

Reply:
Thanks for breaking this down @mitchellhenke and @aduth. Appropriate changes *should* be up